New feature: mu4e-compose-signature

cit opened the pull request:
Add `mu4e-compose-signature` to `multimu4e--account-variables`. Add a
function that changes the signature of a message.

cit committed:
New feature: mu4e-compose-signature

Add `mu4e-compose-signature` to `multimu4e--account-variables`. Add a
function that changes the signature of a message.

DamienCassou commented:
Great, I love it. Please add unit tests and I will merge it. Just make sure you try with different kinds of signature (one or two lines) and different number of lines between the text before the signature and the signature.

cit committed:
Improved `multimu4e--change-signature-in-compose`

This function now detects if the previous signature had a blank line
between the ending and the signature. If there was no blank line,
`multimu4e--change-signature-in-compose` removes the blank line before
the new signature.

cit committed:
Added unit tests for `multimu4e--change-signature-in-compose`

cit commented:
Do you have any ideas why the test fails? On my system with emacs 24.5.1 all tests passed:

```
Selector: t
Passed:  16
Failed:  0
Skipped: 0
Total:   16/16

Started at:   2015-08-04 09:03:09+0200
Finished.
Finished at:  2015-08-04 09:03:09+0200
```

DamienCassou commented on multimu4e.el:
```diff
@@ -180,6 +181,38 @@ If NAME or ADDRESS are not provided, use the variables `user-full-name' and
+      ;; If the current message has a signature, delete it first
+      (if (message-goto-signature)
+          (progn
+            (previous-line 2)
```
in the docstring of `previous-line`, you will read:

```
If you are thinking of using this in a Lisp program, consider using
`forward-line' with a negative argument instead.  It is usually easier
to use and more reliable (no dependence on goal column, etc.).
```

DamienCassou commented on multimu4e.el:
```diff
@@ -180,6 +181,38 @@ If NAME or ADDRESS are not provided, use the variables `user-full-name' and
+      (if last-line-not-empty
+          (progn
+            (message-goto-signature)
+            (previous-line 1)
```
same thing

DamienCassou commented on test/multimu4e-tests.el:
```diff
@@ -116,6 +116,35 @@
+    (setq mu4e-compose-signature "new signature")
+    (multimu4e--change-signature-in-compose)
+    (message-goto-signature)
+    (previous-line 3)
```
same thing

DamienCassou commented on test/multimu4e-tests.el:
```diff
@@ -116,6 +116,35 @@
+    (setq mu4e-compose-signature "new signature")
+    (multimu4e--change-signature-in-compose)
+    (message-goto-signature)
+    (previous-line 2)
```
same thing

DamienCassou commented on multimu4e.el:
```diff
@@ -180,6 +181,38 @@ If NAME or ADDRESS are not provided, use the variables `user-full-name' and
+the signature, this function removes the blank line before the
+new signature."
+  (save-excursion
+    (set (make-local-variable 'message-signature) mu4e-compose-signature)
```
could you please add a #'boundp check before using `mu4e-compose-signature`? Your code should work beyond mu4e.

DamienCassou commented:
Could you please move your patch to https://github.com/DamienCassou/profile ? I'm deprecating multimu4e as the exact same code can work for other MUA.

DamienCassou commented:
Florian Adamsky notifications@github.com writes:

> Do you have any ideas why the test fails? On my system with emacs 24.5.1 all tests passed:

the tests also pass on my computer... As soon as you apply the changes I
talked about before (most importantly the one about
DamienCassou/profile), I will merge your changes and take care of this
error (don't know how yet :-)).

## 

Damien Cassou
http://damiencassou.seasidehosting.st

"Success is the ability to go from one failure to another without
losing enthusiasm." --Winston Churchill
